src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5447,6 +5447,18 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_DisallowNullAttributeForbidsMaybeNullAssignment_Title" xml:space="preserve">
     <value>A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]</value>
   </data>
+  <data name="WRN_ParameterConditionallyDisallowsNull" xml:space="preserve">
+    <value>Parameter '{0}' may not have a null value when exiting with '{1}'.</value>
+  </data>
+  <data name="WRN_ParameterConditionallyDisallowsNull_Title" xml:space="preserve">
+    <value>Parameter may not have a null value when exiting in some condition.</value>
+  </data>
+  <data name="WRN_MethodShouldThrow" xml:space="preserve">
+    <value>A method marked with [DoesNotReturn] should throw on all code paths.</value>
```

Comment:
>should throw on all code paths [](start = 48, length = 30)

Is the following valid?
```C#
[DoesNotReturn]
static int F()
{
    while (true) { }
}
```

Reply:
➡️ added test (no warning on above method)

Reply:
Sorry, I wasn't clear. I meant a method marked `[DoesNotReturn]` does not need to throw on all code paths. Perhaps the message should be "A method marked [DoesNotReturn] should not return."

---
In reply to: [369386905](https://github.com/dotnet/roslyn/pull/41098#discussion_r369386905) [](ancestors = 369386905)